api/addons/serializers.py

```diff
@@ -1,10 +1,15 @@
 from rest_framework import serializers as ser
 from api.base.serializers import JSONAPISerializer, LinksField
 from api.base.utils import absolute_reverse
+from distutils.version import StrictVersion
 
 class NodeAddonFolderSerializer(JSONAPISerializer):
     class Meta:
-        type_ = 'node_addon_folders'
+        @staticmethod
+        def get_type(request):
+            if StrictVersion(request.version) < StrictVersion('2.15'):
+                return 'node_addon_folders'
+            return 'node-addon-folders'
```

Comment:
Consider adding a method that you can call inside every get_type(request), that takes in the old type and the new type, and returns the correct type depending on if the version is less than 2.16.  I'd like to DRY up all of these ` if StrictVersion(request.version) < StrictVersion('2.15')`